### PR TITLE
New 4D Variable Structure: nlayers instead of nlev; starting to adjust docs

### PR DESCRIPTION
**Pull request by maximilian-gelbrecht**
This belongs and points to #525  

This turned out to be more than just the docs, so it's a separate branch for now. 

* I changed `nlev` to `nlayers` (this was needed so that e.g. the analysis script works)
* adjusted the Callbacks somewhat
* Went through the documentation and updated occurrences of `nlev` and the old structure of `PrognosticVariables` and `DiagnosticVariables` , I also made sure e.g. the analaysis and callback docs actually run.

Let's see how the doc CI does, I only run the tests locally. 

The first error in the tests is about the netCDF output which also has to be adjusted definitely a bit more to the new structure.

### CODE REVIEW

File: docs/src/callbacks.md [hunk 1]
```diff
@@ -74,8 +74,9 @@ function SpeedyWeather.initialize!(
     callback.maximum_surface_wind_speed = zeros(progn.clock.n_timesteps + 1)
     
     # where surface (=lowermost model layer) u, v on the grid are stored
-    (; u_grid, v_grid) = diagn.layers[diagn.nlev].grid_variables
-    
+    u_grid = diagn.grid.u_grid[:,diagn.nlayers]
```
Comment (milankl) [inline docs/src/callbacks.md:77]: Can you, in general, put a space ` ` after every `,` or `;` ?

File: docs/src/analysis.md [hunk 2]
```diff
@@ -401,10 +401,10 @@ function global_diagnostics(u, v, ζ, η, model)
     
     h = @. η + H - Hb           # thickness
     q = @. (ζ + f) / h          # potential vorticity
-    λ = @. u * r + Ω * r^2      # angular momentum
-    k = @. 1/2 * (u^2 + v^2)    # kinetic energy
-    p = @. 1/2 * g * h          # potential energy
-    z = @. q^2/2                # potential enstrophy
+    λ = @. NF(u * r + Ω * r^2 )     # angular momentum (in the right number format)
```
[the remark this answers is not in the document]
Comment (maximilian-gelbrecht) [reply docs/src/analysis.md:404]: `R` is `Float64`  and the `1/2` is `Float64` as well. 

All of these let to errors in the `transform` in `∬dA`

**Review by milankl — APPROVED** [review]

File: docs/src/callbacks.md (hunk 1 above)
Comment (milankl) [reply docs/src/callbacks.md]: Just did that already in [4bab9bc](https://github.com/SpeedyWeather/SpeedyWeather.jl/pull/566/commits/4bab9bcdc31001777d09fe31f94f43f6f84102b7)

File: docs/src/analysis.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (milankl) [reply docs/src/analysis.md]: Shall we convert `r` but also do `/ 2` instead, the cleanest way?

[the remark this answers is not in the document]
Comment (milankl) [reply docs/src/analysis.md]: Did that in [95a4f4d](https://github.com/SpeedyWeather/SpeedyWeather.jl/pull/566/commits/95a4f4de8bf882741887758c8bc4afd3d60ac5fd)

**Comment by milankl** [comment]
I'm thinking of merging this PR because the other errors that are now popping up in the documentation are things like `write_netcdf_variables!` which I haven't translated yet to the new structure, so this doesn't really belong here to fix it.

**Comment by maximilian-gelbrecht** [comment]
Yes, please do. When I work on it again, I'll just open up another PR/branch

**Pull request merged**